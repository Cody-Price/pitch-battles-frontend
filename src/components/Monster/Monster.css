@keyframes monster-attack {
  0% {
    left: 150px;
  }
  50% {
    left: 130px;
  }
  100% {
    left: 150px;
  }
}

@keyframes monster-hit {
  0% {
    left: 150px;
  }
  50% {
    left: 170px;
  }
  100% {
    left: 150px;
  }
}

@keyframes monster-dead {
  0% {
    opacity: 0;
  }
  15% {
    opacity: 0;
  }
  15.01% {
    opacity: 1;
  }
  30% {
    opacity: 1;
  }
  30.01% {
    opacity: 0;
  }
  45% {
    opacity: 0;
  }
  45.01% {
    opacity: 1;
  }
  60% {
    opacity: 1;
  }
  60.01% {
    opacity: 0;
  }
  75% {
    opacity: 0;
  }
  75.01% {
    opacity: 1;
  }
  90% {
    opacity: 1;
  }
  90.01% {
    opacity: 0;
  }
  100% {
    opacity: 0;
  }
}
