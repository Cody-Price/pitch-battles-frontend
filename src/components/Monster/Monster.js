import React, { Component } from "react";

import "./Monster.css";
import "./MonsterAnims/Boar.css";
import "./MonsterAnims/Bird.css";
import "./MonsterAnims/Phoenix.css";
import "./MonsterAnims/Minotaur.css";
import "./MonsterAnims/Demon.css";

import monsters from "../../utilities/monsters.js";

class Monster extends Component {
  render() {
    let currentMonster = monsters[this.props.level];

    return <div className={`monster ${currentMonster} ${this.props.status}`} />;
  }
  1;
}

export default Monster;
