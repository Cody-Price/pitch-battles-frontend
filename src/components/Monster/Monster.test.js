import React from "react";
import ReactDOM from "react-dom";
import Monster from "./Monster";

it("is a test", () => {});
