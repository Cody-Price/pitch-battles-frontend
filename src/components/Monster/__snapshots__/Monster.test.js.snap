// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monster should match the snapshot 1`] = `
<div
  className="monster undefined"
  style={
    Object {
      "backgroundImage": "url(./spritesheets/monsters/monsterundefined.png)",
    }
  }
/>
`;
